msdk-id-sirius/src/main/java/NativeLibraryLoader.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * (C) Copyright 2015-2017 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+
+import com.google.common.io.Files;
+import io.github.msdk.MSDKException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+public class NativeLibraryLoader {
+  private NativeLibraryLoader() {}
+  private static Path getResourcePath(String resource) throws MSDKException {
+    final URL url = NativeLibraryLoader.class.getClassLoader().getResource(resource);
+    try {
+      return Paths.get(url.toURI()).toAbsolutePath();
+    } catch (URISyntaxException e) {
+      throw new MSDKException(e);
+    }
+  }
+
+  public static void loadLibraryFromJar(String folder, String[] libs) throws MSDKException, IOException {
+    String suffix = "", realPath;
+    String arch = System.getProperty("os.arch").toLowerCase().endsWith("64") ? "w64" : "w32";
```

Comment:
Check here for the possible os.name and os.arch values: http://www.java-gaming.org/index.php/topic,14110